rpc/dial.go

```diff
@@ -93,90 +108,161 @@ func dial(
+				// This path is also called by an mdns direct connection and ignores that case.
+				// This will skip all Authenticate/AuthenticateTo calls for the signaler.
+				if !dOpts.usingMDNS && dOpts.authMaterial == "" && dOpts.webrtcOpts.SignalingExternalAuthAuthMaterial != "" {
+					logger.Debug("using signaling's external auth as auth material")
+					dOpts.authMaterial = dOpts.webrtcOpts.SignalingExternalAuthAuthMaterial
+					dOpts.creds = Credentials{}
+				}
 			}
-			fixupWebRTCOptions(dOpts, target, port)
-
-			// When connecting to an external signaler for WebRTC, we assume we can use the external auth's material.
-			// This path is also called by an mdns direct connection and ignores that case.
-			// This will skip all Authenticate/AuthenticateTo calls for the signaler.
-			if !dOpts.usingMDNS && dOpts.authMaterial == "" && dOpts.webrtcOpts.SignalingExternalAuthAuthMaterial != "" {
-				logger.Debug("using signaling's external auth as auth material")
-				dOpts.authMaterial = dOpts.webrtcOpts.SignalingExternalAuthAuthMaterial
-				dOpts.creds = Credentials{}
+
+			if dOpts.debug {
+				logger.Debugw(
+					"trying WebRTC",
+					"signaling_server", dOpts.webrtcOpts.SignalingServerAddress,
+					"host", originalAddress,
+				)
 			}
-		}
 
-		if dOpts.debug {
-			logger.Debugw(
-				"trying WebRTC",
-				"signaling_server", dOpts.webrtcOpts.SignalingServerAddress,
-				"host", originalAddress,
-			)
-		}
+			conn, cached, err := dialFunc(
+				ctxParallel,
+				"webrtc",
+				fmt.Sprintf("%s->%s", dOpts.webrtcOpts.SignalingServerAddress, originalAddress),
+				buildKeyExtra(dOpts),
+				func() (ClientConn, error) {
+					return dialWebRTC(
+						ctxParallel,
+						dOpts.webrtcOpts.SignalingServerAddress,
+						originalAddress,
+						dOpts,
+						logger,
+					)
+				})
+
+			switch {
+			case err == nil:
+				if !cached {
+					logger.Debug("connected via WebRTC")
+				} else if dOpts.debug {
+					logger.Debug("connected via WebRTC (cached)")
+				}
+				dialCh <- dialResult{conn: conn, cached: cached}
+			case !errors.Is(err, ErrNoWebRTCSignaler):
+				// TODO(RSDK-6493): Investigate if we must `skipDirect` here.
+				dialCh <- dialResult{err: err, skipDirect: true}
+			case ctxParallel.Err() != nil:
+				dialCh <- dialResult{err: ctxParallel.Err(), skipDirect: true}
+			default:
+				dialCh <- dialResult{err: err}
+			}
+		}()
+	}
 
-		conn, cached, err := dialFunc(
-			ctx,
-			"webrtc",
-			fmt.Sprintf("%s->%s", dOpts.webrtcOpts.SignalingServerAddress, originalAddress),
-			buildKeyExtra(dOpts),
-			func() (ClientConn, error) {
-				return dialWebRTC(
-					ctx,
-					dOpts.webrtcOpts.SignalingServerAddress,
-					originalAddress,
-					dOpts,
-					logger,
-				)
-			})
-		if err == nil {
-			if !cached {
-				logger.Debug("connected via WebRTC")
-			} else if dOpts.debug {
-				logger.Debug("connected via WebRTC (cached)")
+	go func() {
```

Comment:
```suggestion
        // Make sure the slower connection attempt is fully cancelled, or if the attempt succeeded,
        // close the slower connection.
	go func() {
```
can we add a comment here to reference its purpose in the overall function?